src/pages/Transfer/CrossChainTransferForm/CrossChainTransferForm.tsx

```diff
@@ -35,125 +43,255 @@ import {
+    if (!accountId) return;
+
+    form.setFieldValue(CROSS_CHAIN_TRANSFER_TO_FIELD, e.target.value);
+
+    const tokens = await getAvailableTokens(
+      form.values[CROSS_CHAIN_TRANSFER_FROM_FIELD] as ChainName,
+      e.target.value as ChainName,
+      accountId.toString(),
+      form.values[CROSS_CHAIN_TRANSFER_TO_ACCOUNT_FIELD] as string
+    );
+
+    if (!tokens) return;
 
-    form.setFieldValue(CROSS_CHAIN_TRANSFER_TO_ACCOUNT_FIELD, accountId.toString());
-    form.validateField(CROSS_CHAIN_TRANSFER_AMOUNT_FIELD);
+    setTransferableTokens(tokens);
+    setCurrentToken(tokens[0]);
+  };
+
+  useEffect(() => {
+    if (!transferableTokens.length) return;
+    const defaultToken = transferableTokens[0];
+
+    form.setFieldValue(CROSS_CHAIN_TRANSFER_TOKEN_FIELD, defaultToken.value);
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [accountId]);
+  }, [transferableTokens]);
 
-  const transferMonetaryAmount = newSafeMonetaryAmount(
-    form.values[CROSS_CHAIN_TRANSFER_AMOUNT_FIELD] || 0,
-    GOVERNANCE_TOKEN,
-    true
-  );
+  const handleTickerChange = (ticker: string, name: string) => {
+    form.setFieldValue(name, ticker);
+    setCurrentToken(transferableTokens.find((token) => token.value === ticker));
+  };
+
+  const handleDestinationAccountChange: ChangeEventHandler<HTMLInputElement> = (e) => {
+    form.setFieldValue(CROSS_CHAIN_TRANSFER_TO_ACCOUNT_FIELD, e.target.value);
+  };
+
+  const transferMonetaryAmount = currentToken
+    ? newSafeMonetaryAmount(
+        form.values[CROSS_CHAIN_TRANSFER_AMOUNT_FIELD] || 0,
+        getCurrencyFromTicker(currentToken.value),
+        true
+      )
+    : 0;
+
+  const valueUSD = transferMonetaryAmount
+    ? convertMonetaryAmountToValueInUSD(
+        transferMonetaryAmount,
+        getTokenPrice(prices, currentToken?.value as string)?.usd
+      )
+    : 0;
+
+  const isCTADisabled = isFormDisabled(form) || form.values[CROSS_CHAIN_TRANSFER_AMOUNT_FIELD] === '';
+
+  useEffect(() => {
+    if (!originatingChains?.length) return;
 
-  const valueUSD =
-    convertMonetaryAmountToValueInUSD(transferMonetaryAmount, getTokenPrice(prices, GOVERNANCE_TOKEN.ticker)?.usd) ?? 0;
+    // This prevents a render loop caused by setFieldValue
+    if (form.values[CROSS_CHAIN_TRANSFER_FROM_FIELD]) return;
 
-  const isCTADisabled = isFormDisabled(form);
+    const destinationChains = getDestinationChains(originatingChains[0].id as ChainName);
+
+    form.setFieldValue(CROSS_CHAIN_TRANSFER_FROM_FIELD, originatingChains[0].id);
+    form.setFieldValue(CROSS_CHAIN_TRANSFER_TO_FIELD, destinationChains[0].id);
+
+    setDestinationChains(destinationChains);
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [originatingChains]);
+
+  useEffect(() => {
+    if (!destinationChains?.length) return;
+    if (!accountId) return;
+
+    const getTokensForNewChain = async () => {
+      const tokens = await getAvailableTokens(
```

Comment:
There are a lot of type assertions on this file. Arent they possible to be solved?

Reply:
Done what I can here—think we can improve this further at some point by handling types better across the XCM bridge api/UI